python/lsst/pipe/tasks/transformMeasurement.py

```diff
@@ -152,38 +138,85 @@ class RunTransformTask(pipeBase.CmdLineTask):
+    @param[in] measConfigType  Name of measurement task configuration (e.g. processCCd_config, etc)
+    @param[in] outputType      Name of output type to write (e.g. transformedSrc, etc)
+    @param[in] wasForced       Boolean indicating if source measurements were forced.
+
+    @returns A tuple of (Config, CmdLineTask) suitable for transforming sources of the given type.
+    """
+    sourceTypeField = pexConfig.Field(dtype=str, default=sourceType,
+                                      doc="Dataset type of source to transform")
+    calexpTypeField = pexConfig.Field(dtype=str, default=calexpType,
+                                      doc="Dataset type of calibration exposure")
+    measConfigField = pexConfig.Field(dtype=str, default="processCcd_config",
+                                      doc="Dataset type of measurement operation configuration")
+    outputTypeField = pexConfig.Field(dtype=str, default=outputType,
+                                      doc="Dataset type of output sources")
+    forcedField = pexConfig.Field(dtype=bool, default=wasForced, doc="Measurement operation was forced")
+
+    configClass = type("%sTransformConfig" % sourceType.capitalize(),
+                       (RunTransformConfigBase, pexConfig.Config),
+                       {"sourceType": sourceTypeField, "calexpType": calexpTypeField,
+                        "outputType": outputTypeField, "measConfig": measConfigField,
+                        "wasForced": forcedField})
+    taskClass = type("%sTransformTask" % (sourceType.capitalize(),),
+                     (RunTransformTaskBase, pipeBase.CmdLineTask),
+                     {"wasForced": wasForced, "ConfigClass": configClass,
+                      "_DefaultName": "transform%sMeasurement" % (sourceType.capitalize(),)})
+    return configClass, taskClass
+
+
+# Explicitly instantiate command line transformation types for common source types. Camera-specific
+# transformation tasks can be defined following this pattern in the appropriate obs_ package.
+SrcTransformConfig, SrcTransformTask = makeTransformCmdLineTask(
+                                                "src", "calexp", "processCcd_config", "transformedSrc", False)
+ForcedSrcTransformConfig, ForcedSrcTransformTask = makeTransformCmdLineTask(
+                                       "forced_src", "calexp", "forced_config", "transformedForced_src", True)
+DeepCoaddSrcTransformConfig, DeepCoaddSrcTransformTask = makeTransformCmdLineTask(
+           "deepCoadd_src", "deepCoadd_calexp", "deep_processCoadd_config", "transformedDeepCoadd_src", False)
+DeepCoaddForcedSrcTransformConfig, DeepCoaddForcedSrcTransformTask = makeTransformCmdLineTask(
+                                        "deepCoadd_forced_src", "deepCoadd_calexp", "deepCoadd_forced_config",
+                                                                      "transformedDeepCoadd_Forced_src", True)
```

Comment:
I don't think we want separate tasks for different kinds of coadds (e.g. "deepCoadd" vs "goodSeeingCoadd").  That needs to be something set via config (see my proposal on the main conversation section of the PR).